### PR TITLE
add configurable instances for date, time and datetime

### PR DESCRIPTION
**Pull request by melrief**
@leifwickland @jcazevedo an initial attempt to implement #65. I have implemented configurable instances for java 8's `LocalDate`, `LocalTime` and `LocalDateTime`. What is missing but I would like to do in another PR because I don't have time right now:

1. `TimeZone` and `Period` are not yet covered but they are easy to add because not configurable. I'll do that in another PR
2. `OffsetTime` and `OffsetDateTime` because I never used them. As above, another PR is the right place to put them, I think

I would like to add everything and add a section to the README.md before we release the next version.

### CODE REVIEW

**Comment by jcazevedo** [comment]
This looks good to me! 

We should probably try to normalize the method names of some of the helpers we provide, though. Right now we have `fromString`, `fromNonEmptyString`, `stringConvert` and `nonEmptyStringConvert` as helpers to build `ConfigConvert` from `String`; some of the implicit `ConfigConvert` we provide are named `read<type>` while others are named `<type>ConfigConvert` and this `make<type>Instance` seems to introduce another naming style.

**Comment by melrief** [comment]
@jcazevedo you are right and I would prefer to use the `ConfigConvert` but I don't have a strong opinion about it. What do you prefer?

**Comment by jcazevedo** [comment]
I don't have a strong preference towards a specific alternative. I think I also like the `<type>ConfigConvert` names, so that we would have `stringConfigConvert`, `intConfigConvert`, `localDateConfigConvert(...)`, etc.

**Comment by leifwickland** [comment]
I'll second the idea of using `localDateConfigConvert`, etc for the naming convention.

File: core/src/main/scala/pureconfig/configurable/package.scala
Comment (leifwickland) [inline core/src/main/scala/pureconfig/configurable/package.scala:21]: I aspire to add a `pureconfig-joda` JAR once this gets merged. 
@melrief Would you expect that the similar methods in that JAR would go in a `pureconfig.joda.configurable` package or would you have them load them into this namespace?
I ask more from a philosophical standpoint of how we want to structure the library's API. I don't really have a good answer to that question.

Comment (leifwickland) [inline core/src/main/scala/pureconfig/configurable/package.scala:21]: Do you want to add a `/** ... */` comment declaring the purpose of this package? At first glance, it'd seem to be limited to date/time types, but I suspect you intend it more broadly than that.

**Review by leifwickland — APPROVED** [review]
👍 
Thanks for doing this, @melrief!
Are you planning to add a README section about this?

File: core/src/main/scala/pureconfig/configurable/package.scala
Comment (melrief) [reply core/src/main/scala/pureconfig/configurable/package.scala:21]: I think the behavior of modules should mimic the one of core as much as possible, so yes, for me it should go in `pureconfig.joda.configurable`.

Comment (melrief) [inline core/src/main/scala/pureconfig/configurable/package.scala]: We'll do

Comment (melrief) [reply core/src/main/scala/pureconfig/configurable/package.scala:21]: Yes, I'll do it

**Comment by melrief** [comment]
@leifwickland @jcazevedo because there are some conflicts with master, I'll rebase this branch on master and force push it.

File: core/src/main/scala/pureconfig/configurable/package.scala
Comment (leifwickland) [inline core/src/main/scala/pureconfig/configurable/package.scala]: Nitpick: I think 
"... takes an input [[java.time.format.DateTimeFormatter]] ..."
would read more naturally.

File: core/src/main/scala/pureconfig/configurable/package.scala
Comment (leifwickland) [inline core/src/main/scala/pureconfig/configurable/package.scala:13]: I think it might be useful to try to explain where the user of the library needs to include this line of code, as is always the case with Scala's implicits.

**Comment by melrief** [comment]
@leifwickland I've added a section of the README.md, you can see it [here](https://github.com/melrief/pureconfig/tree/add_local_datetime_instances#configurable-converters). I wasn't sure how to explain it properly, let me know if it's understandable.

**Comment by leifwickland** [comment]
@melrief I think your addition to the README makes sense. I can't think of a way to improve it beside the nitpick I mentioned. 👍

**Comment by melrief** [comment]
@jcazevedo you had a look at his PR before, what do you think of my changes?

**Review by jcazevedo — APPROVED** [review]
I think this is good to merge.

I'm not particularly fond of having these factory methods prefixed with `make`, but I'm OK with it as long as we're consistent with this naming convention in future configurable instances.

**Comment by melrief** [comment]
@jcazevedo we can remove the `make` prefix for configurable instances, I'm not particularly fond of the prefix anyway. @leifwickland @ruippeixotog what do you think about it? I prefer to decide now for the future instead of having to refactor the configurable instances another time.

**Comment by ruippeixotog** [comment]
I support the `<type>ConfigConvert` naming 👍

**Comment by leifwickland** [comment]
I'm all for removing the `make` prefix.

**Comment by melrief** [comment]
Removed, thanks guys! Waiting for CI to run the tests and then I think we can merge.

**Pull request merged**
